### PR TITLE
Supporting mixins

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**

### CODE REVIEW

File: src/main/java/com/google/api/codegen/go/GoGapicContext.java [hunk 1]
```diff
@@ -128,7 +129,12 @@ public String getClientPrefix(Interface service) {
     String name = getReducedServiceName(service);
     // If there's only one service, or the service name matches the package name, don't prefix with
     // the service name.
-    if (getModel().getSymbolTable().getInterfaces().size() == 1 || name.equals(getPackageName())) {
+    int interfaceCount = 0;
+    for (@SuppressWarnings("unused")
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/go/GoGapicContext.java]: Maybe use Iterables.size()?

Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/go/GoGapicContext.java]: Good idea, will do.

File: src/main/java/com/google/api/codegen/py/PythonImportHandler.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public PythonImportHandler(Interface service) {
     // Add method request-type imports.
     for (Method method : service.getMethods()) {
       addImport(
-          method.getFile(),
+          method.getInputMessage().getFile(),
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/py/PythonImportHandler.java]: Why is this required? Is it just because of the updated api compiler version?

Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/py/PythonImportHandler.java]: The original is actually incorrect, but the problem didn't surface until I added mixin support. Basically, with mixins, the file declaring the input message type can be distinct from the file declaring the RPC; the type here needs to match the file of the input message type.

**Comment by garrettjonesgoogle** [comment]
I have addressed the feedback so far.

File: src/main/java/com/google/api/codegen/config/ConfigGeneratorApi.java [hunk 3]
```diff
@@ -15,9 +15,9 @@
 package com.google.api.codegen.config;
 
 import com.google.api.client.util.Lists;
-import com.google.api.codegen.CodegenSetup;
 import com.google.api.codegen.ConfigProto;
 import com.google.api.tools.framework.aspects.http.model.HttpAttribute;
+import com.google.api.tools.framework.aspects.mixin.model.MixinAttribute;
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/config/ConfigGeneratorApi.java]: Remove unused imports

Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/config/ConfigGeneratorApi.java]: I blame Eclipse. Regardless, I have fixed it now.

File: src/main/java/com/google/api/codegen/go/GoGapicContext.java [hunk 4]
```diff
@@ -544,7 +547,7 @@ public boolean hasEmptyRetryCodes(Interface service, MethodConfig methodConfig)
    * Returns true if the API has a page streaming method.
    */
   public boolean hasPageStreamingMethod() {
-    for (Interface service : getModel().getSymbolTable().getInterfaces()) {
+    for (Interface service : new InterfaceView().getElementIterable(getModel())) {
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/go/GoGapicContext.java]: It would be nice if the `new InterfaceView().getElementIterable(model)` pattern could be a method somewhere, but I'm not sure if there is anywhere sensible...

Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/go/GoGapicContext.java]: I couldn't think of anything that would be a significant improvement over what I have here.

**Comment by michaelbausor** [comment]
LGTM
